### ISSUE TITLE
Add "Use GitHub.com Style" functionality

### ISSUE THREAD

**Issue by thomasjo** [atom/markdown-preview#244]
_NOTE: This is a tracking issue spawned out of #240_

Since it's not unlikely that most users previewing their markdown files in Atom will end up pushing them to github.com, we should provide an option that will ensure the previewed markdown document closely resembles what it would look like if rendered on github.com, instead of using the current UI theme's (markdown) style. Not all themes override the markdown-preview styles, but our default UI theme (One Dark) does, and this functionality gives users the option to enforce a consistent preview experience regardless of possible UI theme trickery.
- [x] Finish, and ship #240.
- [ ] Use GitHub.com style for fenced code blocks.
  - [ ] Solve by creating/using a GitHub.com syntax theme for Atom?
- [x] Investigate automating the creation of the GitHub markdown styles to better future-proof the functionality.
  - [x] Possibility of getting non-minified style from GitHub? **Solved via: https://github.com/primer/markdown/blob/master/components/markdown.scss**
  - [ ] Automate the process of keeping the stylesheet (and possible syntax theme) up to date.

**Comment by wufeifei** [comment]
👍

**Comment by aioue** [comment]
+1

**Comment by negativefix** [comment]
:thumbsup:

**Comment by enyusmile** [comment]
good job!

**Comment by ericoporto** [comment]
When in a list, (with `1. ` or  `* `), you can place a code block inside and item by inserting a new line and then indenting with **8 spaces** . BUT in the **Markdown Preview**, it uses 8 spaces for the first new line and 7 spaces for the following lines.

**Comment by simurai** [comment]
Carried over from this comment https://github.com/atom/markdown-preview/issues/302#issuecomment-142180584

I think to make code blocks 100% GitHub style, they would need the same markup (with `pre code`) so the `.highlight` styles work. Not sure what GitHub uses for the syntax highlighting, Pygments maybe?

Using `pre code` instead of `atom-text-editor` might also fix issues with "scroll past end" #209 or the scrollbar covering the last line #286.

**Comment by mcleary** [comment]
Maybe this is a silly question but on Mac why the option to use GitHub style does not appear?

**Comment by simurai** [comment]
@mcleary Are you able to see the rest of the package settings? It should be the last option:

![screen shot 2015-09-30 at 11 16 33 pm](https://cloud.githubusercontent.com/assets/378023/10195630/8fd2628c-67c9-11e5-8b13-9f05368e7f6e.png)

**Comment by mcleary** [comment]
Oh, shame on me!. 

The reason that I made this question is I was looking for a way to preview with a darker background.

I thought the solution would be active the GitHub Style option but later I see a way to customize the previewer CSS. Is the recommended way of doing so?

**Comment by simurai** [comment]
Preview with darker background already got merged #298 but it isn't released yet.

If you wanna make small changes, you can do them in your `styles.less` file. For example:

```css
.markdown-preview.markdown-preview {
  background-color: #444;
}
```

**Comment by mcleary** [comment]
Thanks, already did that.

I'll wait for #298 release.

**Comment by wiseman** [comment]
Just curious what the status of this is?

**Comment by thomasjo** [comment]
With the exception of styling fenced code blocks, this shipped as part of #240, and was subsequently improved upon, and tweaked in #298.

**Comment by ramtinsoltani** [comment]
I'd like to add two issues to this (not sure if this is the proper place to mention them, so guide me if I'm wrong).

---

First, if the first bullet in a bullet list starts with `1.`, the rest of the bullets will be rendered in Atom as numbers regardless of the syntax.

For example:

```
1. Item 1
- Item 2
* Item 3
```

In Atom:
![image](https://user-images.githubusercontent.com/7918069/31305995-380d87c6-aafb-11e7-961a-8d3a54450f82.png)

In GitHub:

1. Item 1
- Item 2
* Item 3

---

The second thing is inserting a table inside a bullet list. Consider the following (the table is indented by two spaces):

```
- Item 1
- Item 2:

  | Column 1 | Column 2 |
  |------------|------------|
  | Field 1 | Field 2 |
  | Field 1 | Field 2 |
```

In Atom:
![image](https://user-images.githubusercontent.com/7918069/31306045-8b91eb16-aafc-11e7-9fa3-b23bcd464c6c.png)

In GitHub:

- Item 1
- Item 2:

  | Column 1 | Column 2 |
  |------------|------------|
  | Field 1 | Field 2 |
  | Field 1 | Field 2 |
